spec/lang/step.md

```diff
@@ -558,13 +558,13 @@ impl<M: Memory> Machine<M> {
         // Evaluate all arguments.
         let arguments = arguments.try_map(|arg| self.eval_value(arg))?;
 
-        let (value, ret_type) = self.eval_intrinsic(intrinsic, arguments)?;
+        let (value, _ret_ty) = self.eval_intrinsic(intrinsic, arguments)?;
```

Comment:
I didn't realize this means we entirely ignore the `ret_ty` here. That's not good. Argh why are function calls so annoying.^^ I'll have to think about what we want to do here.

We should definitely add a test for calling an intrinsic where the result is put into a place of the wrong type.

Reply:
Isn't it the case that we also don't check anything about types when a function returns? I have just written a test where a u64 is returned into a u8 and it passes.

Reply:
That sounds potentially concerning, could you open an issue and paste the example there?